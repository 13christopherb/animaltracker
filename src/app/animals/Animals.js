import React, {Component} from 'react';
import {connect} from 'react-redux';
import { actions } from './ducks/actions';
import { AnimalRow } from './AnimalRow';


class Animals extends Component {

    constructor(props) {
        super(props);
        this.state = {value: ''}
        this.deleteAnimal = this.deleteAnimal.bind(this)
    }

    componentDidMount() {
    }

    deleteAnimal(animal) {
        this.props.dispatch(actions.deleteAnimal(animal.props.animal));
    }

    render() {
        let animals = [];
        for (let animal of this.props.animals) {
            animals.push(<AnimalRow animal={animal} deleteAnimal={this.deleteAnimal} key={animal.id} />);
        }
        return (
            <div>
                <div className="row">
                    <div className="col col-md-12">
                        <div className="table-responsive">
                            <table className="table table-striped">
                                <tbody>
                                {animals}
                                </tbody>
                            </table>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}

function mapStateToProps({animals}, ownProps) {
    return {
        animals: ownProps.animals
    }
}

const connectedAnimals = connect(
    mapStateToProps,
)(Animals);

export {connectedAnimals as Animals}
