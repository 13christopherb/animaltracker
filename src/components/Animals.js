import React, {Component} from 'react';
import {connect} from 'react-redux';
import uuidv4 from 'uuid';
import * as actions from '../actions/actions';
import AnimalTitle from './AnimalTitle';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'


class Animals extends Component {

    constructor(props) {
        super(props);
        this.state = {value: ''}
    }

    componentDidMount() {
        this.props.dispatch(actions.getAnimals());
    }



    render() {
        let animals = [];
        for (let animal of this.props.animals) {
            animals.push(<AnimalTitle animal={animal} key={animal.id} />);
        }
        return (
            <div>
                <div className="row">
                    <table className="table table-striped">
                        <tbody>
                        {animals}
                        </tbody>
                    </table>
                </div>
            </div>
        );
    }
}

function mapStateToProps({animals}, ownProps) {
    return {
        animals: animals
    }
}

export default connect(
    mapStateToProps,
)(Animals)
