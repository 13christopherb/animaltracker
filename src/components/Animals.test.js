import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import Animals from './Animals';
import toJson from 'enzyme-to-json';
import thunk from 'redux-thunk'

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)

const initialState = {
    animals: [
        {
            name: 'Edginald',
            species: 'CSL',
            weight: 14,
            isGettingTubed: true,
            isGettingControlledMeds: false,
            id: 1
        },
        {
            name: 'Pipester',
            species: 'NFS',
            weight: 8,
            isGettingTubed: true,
            isGettingControlledMeds: false,
            id: 2
        },
        {
            name: 'Kirt',
            species: 'CSL',
            weight: 85,
            isGettingTubed: false,
            isGettingControlledMeds: true,
            id: 3
        }

    ]
}

const store = mockStore(initialState)

describe('<Animals />', () => {
    describe('render()', () => {
        test('should render the component', () => {
        const wrapper = shallow(<Animals store={store} />)
        const component = wrapper.dive()

        expect(toJson(component)).toMatchSnapshot()
        })
    })
});