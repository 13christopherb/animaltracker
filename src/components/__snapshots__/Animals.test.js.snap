// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Animals /> render() should render the component 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col col-md-12"
    >
      <div
        className="table-responsive"
      >
        <table
          className="table table-striped"
        >
          <tbody>
            <AnimalTitle
              animal={
                Object {
                  "id": 1,
                  "isGettingControlledMeds": false,
                  "isGettingTubed": true,
                  "name": "Edginald",
                  "species": "CSL",
                  "weight": 14,
                }
              }
              deleteAnimal={[Function]}
              key="1"
            />
            <AnimalTitle
              animal={
                Object {
                  "id": 2,
                  "isGettingControlledMeds": false,
                  "isGettingTubed": true,
                  "name": "Pipester",
                  "species": "NFS",
                  "weight": 8,
                }
              }
              deleteAnimal={[Function]}
              key="2"
            />
            <AnimalTitle
              animal={
                Object {
                  "id": 3,
                  "isGettingControlledMeds": true,
                  "isGettingTubed": false,
                  "name": "Kirt",
                  "species": "CSL",
                  "weight": 85,
                }
              }
              deleteAnimal={[Function]}
              key="3"
            />
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
